framework/test/mocha/unit/modules/chain/helpers/config.js

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const config = require('../../../../../../src/modules/chain/helpers/config');
+const {
+	ZSchema,
+} = require('../../../../../../src/controller/helpers/validator');
+
+describe('helpers/config', () => {
+	const packageJsonStub = {
+		version: '1.5.0',
+		lisk: {
+			protocolVersion: '1.0',
+		},
+	};
+
+	const env = Object.assign({}, process.env);
+	const baseArgs = [
+		'/Users/michiel.mulders/.nvm/versions/node/v10.15.3/bin/node',
+		'/Users/michiel.mulders/lisk/lisk/src/index.js',
+	];
+
+	beforeEach(async () => {
+		process.argv = [];
+
+		ZSchema.getLastErrors = sinonSandbox.stub();
+		process.exit = sinonSandbox.stub();
+		console.error = sinonSandbox.stub();
+	});
+
+	afterEach(async () => {
+		sinonSandbox.restore();
+		process.env = env;
+	});
+
+	it('should return appConfig containing correct properties', async () => {
+		const appConfig = config(packageJsonStub, false);
+
+		// Only check if most important properties are present
+		expect(appConfig).to.have.property('db');
+		expect(appConfig).to.have.property('redis');
+		expect(appConfig).to.have.property('api');
+		expect(appConfig).to.have.property('peers');
+		expect(appConfig).to.have.property('genesisBlock');
+		expect(appConfig).to.have.property('constants');
+	});
+
+	describe('Commandline options: process.argv', () => {
+		it('should parse commandline options and return appConfig', async () => {
+			const httpPort = 9999;
+			process.argv = [...baseArgs, '-h', `${httpPort}`];
```

Comment:
why not passing httpPort directly? Does it have to be string? In that case maybe we could use `httpPort.toString()` to avoid someone adding a value next to it since we are passing each parameter as array value?

Reply:
Must be string yes as that's how it is received when you test it as well.